backend/src/handlers_prelude/repo_fs.rs

```diff
@@ -0,0 +1,284 @@
+//! Endpoints for interacting with the repository's filesystem (create doc/asset, read doc/asset, et cetera)
+use crate::git::INode;
+use axum::{
+    body::Bytes,
+    debug_handler,
+    extract::{DefaultBodyLimit, Path, Query, State},
+    http::{HeaderMap, StatusCode},
+    response::IntoResponse,
+    routing::get,
+    Json, Router,
+};
+use reqwest::header::{CONTENT_DISPOSITION, CONTENT_TYPE};
+use serde::{Deserialize, Serialize};
+use tracing::{error, warn};
+
+use crate::{perms::Permission, require_perms, AppState};
+
+use super::eyre_to_axum_err;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct GetDocQuery {
+    pub path: String,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct GetDocResponse {
+    pub contents: String,
+}
+
+/// This handler accepts a `GET` request to `/api/doc?path=`.
+/// TODO: refactor to pass it in directly as a url path instead of doing the whole url arguments thing
+pub async fn get_doc_handler(
+    State(state): State<AppState>,
+    Query(query): Query<GetDocQuery>,
+) -> Result<Json<GetDocResponse>, (StatusCode, &'static str)> {
+    match state.git.get_doc(&query.path) {
+        Ok(maybe_doc) => maybe_doc.map_or(
+            Err((
+                StatusCode::NOT_FOUND,
+                "The file at the provided path was not found.",
+            )),
+            |doc| Ok(Json(GetDocResponse { contents: doc })),
+        ),
+        Err(e) => {
+            warn!(
+                "Failed to fetch doc with path: {:?}; error: {:?}",
+                query.path, e
+            );
+            Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Fetch failed, check server logs for more info",
+            ))
+        }
+    }
+}
```

Comment:
Can this be shortened to:

```rust
if let Some(doc) = state.git.get_doc(&query.path) {
        return match doc {
            Ok(doc) => {
                Ok(Json(GetDocResponse { contents: doc }))
            }
            Err(err) => {
                error!("Failed to fetch document with path {:?}. Error: {:?}", err);
                Err((StatusCode::INTERNAL_SERVER_ERROR, "Unable to fetch doc, check server logs"))
            }
        }
    }
```
How do we know the status code is not found if none? It could be a different reason under certain cases I imagine, so we would be returning the wrong error in this case, so might be worth letting the error handler show the true case every time? I could be wrong though, this is only after a quick glance!

Reply:
@Xeckt `get_doc` returns a `Result<Option<String>>`, if no doc is found it's `Ok(None)`, but for any other error it'll be `Err(e)`.